website/src/pages/docs/threat-detection.md

```diff
@@ -176,3 +169,88 @@ No notifications or security events should be emitted.
+
+### What is GOT Poisoning and Function Redirection
+Function redirection is a mechanism that involves a function used normally by the process to be redirected to point to a nefarious function living in a parasite library. One example involves obfuscation of files. Any given malware may need to place files on a system while not allowing those files to be readily viewed. To accomplish this, the functions that read files from a directory, such as readdir, are redirected to a nefarious function contained in a parasite library. The nefarious readdir removes any reference to malware files and the user is unaware of their existence. 
+
+There are number of ways in which function redirection is accomplished. Here we provide a brief definition of two high level approaches for deploying a nefarious function. One, an attacker-controlled, parasitic shared library is forced to be loaded into the target process, and causes any number of functions to be redirected. Two, a parasitic library is loaded when the process starts using dynamic loader functionality to cause an addition library to be loaded along with the required libraries. The simple example described below uses the second approach.
+
+
+### Notify on GOT errors
+The code below is compiled and creates a shared library. The resulting library is intended to be loaded by the dynamic loader and results in the function readdir being redirected to a nefarious readdir function that blocks the view of a defined file.
+
+A nefarious readdir function is defined with default visibility. This causes the dynamic loader to create PLT/GOT entries for the nefarious readdir when this library is loaded. The dynamic linker is configured to preload this library before all other required libraries are loaded. This results in the nefarious readdir being the new default and used for all references in the process being started.
+
+The constructor obtains the original address of readdir. The original readdir is called by the now default nefarious readdir, the results examined for the file to be obfuscated and removed. The result is that a user is not aware of the presence of the obfuscated file.
+
+#### Run the demo
+Edit a file called got-example.c and paste the code below into that file.
+
+Compile the code.
+```
+gcc -fPIC -g -shared got-example.c -o libobf.so
+```
+
+Create a test file. The Example code obfuscates the existence of a called mycreds.
+```
+touch /tmp/mycreds
+```
+
+Execute "ls" to see that the file mycreds exist
+```
+ls /tmp
+```
+
+Execute ls with the parasitic library preloaded
+```
+export LD_PRELOAD="`pwd`/libobf.so"
+ls /tmp
+```
+
+Now the results from /tmp do not include the file mycreds. It has been obfuscated and does not appear to exist.
+
+```
```

Comment:
it would read better if you move the code up to line 197

Reply:
for sure!